rs/crypto/internal/crypto_lib/bls12_381/type/src/lib.rs

```diff
@@ -1988,6 +1988,39 @@ declare_muln_vartime_affine_impl_for!(G2Projective, G2Affine);
 impl_debug_using_serialize_for!(G2Affine);
 impl_debug_using_serialize_for!(G2Projective);
 
+impl G2Affine {
+    /// Deserialize a public key (compressed format only)
+    ///
+    /// This function verifies that the decoded point is within the prime order
+    /// subgroup, and is safe to call on untrusted inputs.
+    ///
+    /// This function is equivalent to `deserialize` but assumes that public keys
+    /// are seen many times and so makes use of a cache to skip the expensive
+    /// checks when keys are seen again
+    pub fn deserialize_public_key<B: AsRef<[u8]>>(bytes: &B) -> Result<Self, PairingInvalidPoint> {
```

Comment:
Mostly a naming/doc question: it seems there is nothing specific regarding being a key here, this should work for any points, the only difference is that it is cached. So I wonder if we should just rename it to `deserialze_cached` or similar to stress what it is really happening here.